### PR TITLE
CANS-253. Caregiver Domain collects Caregiver Name and displays it in header

### PR DESCRIPTION
**Pull request by denys-davydov**
## Description

The main idea is that caregiver name is stored in the state of Domain component when the user enters the name to make it possible to update the name in the Domain's header. When the input field is blurred the caregiver name is stored in the full assessment state in the Assessment component's state.

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Assessment/style.sass [hunk 1]
```diff
@@ -54,6 +54,14 @@
...
   font-weight: bold
 
+.caregiver-name-wrapper
+  padding: 0.4rem
+
+.caregiver-name
+  margin-bottom: 0.7rem
+  margin-left: 0.2rem
+  width: 40%
+
```
Comment (jonbwalker) [inline app/javascript/Application/components/Assessment/style.sass]: Does this width % scale correctly on both large and small monitors?

File: app/javascript/Application/components/Assessment/Domain.test.js [hunk 2]
```diff
@@ -127,5 +131,82 @@ describe('<Domain />', () => {
...
+            const wrapper = mount(domainComponent);
+            const nameInput = wrapper.find('.caregiver-name').at(0);
+
+            // when
+            nameInput.simulate('change', { target: { value: 'Full Name' } });
+
+            // then
+            expect(wrapper.state('caregiverName')).toEqual('Full Name');
+            expect(wrapper.find('ExpansionPanelSummary').text()).toMatch(/Full Name/);
+          });
```
Comment (jonbwalker) [inline app/javascript/Application/components/Assessment/Domain.test.js]: I've been thinking about our testing strategy, and some of our tests are more like functional tests than unit tests. We should have a discussion about it as a team soon.

File: app/javascript/Application/components/Assessment/style.sass (hunk 1 above)
Comment (denys-davydov) [reply app/javascript/Application/components/Assessment/style.sass]: Yep, it takes 40% of the parent element

File: app/javascript/Application/components/Assessment/Domain.test.js (hunk 2 above)
Comment (denys-davydov) [reply app/javascript/Application/components/Assessment/Domain.test.js]: Good, let's discuss it offline

Comment (edgibbs) [reply app/javascript/Application/components/Assessment/Domain.test.js]: 👍

File: app/javascript/Application/components/Assessment/Domain.js [hunk 3]
```diff
@@ -72,6 +74,30 @@ class Domain extends Component {
...
+
+  handleCaregiverNameUpdate = () => {
+    this.props.onCaregiverNameUpdate(this.props.domain.caregiver_index, this.state.caregiverName);
+  };
+
+  renderCaregiverName = () => {
+    return (
+      <div className={'caregiver-name-wrapper'}>
+        <Input
+          bsSize="lg"
```
Comment (jonbwalker) [inline app/javascript/Application/components/Assessment/Domain.js]: What is bsSize?

Comment (denys-davydov) [reply app/javascript/Application/components/Assessment/Domain.js]: Look here 
https://reactstrap.github.io/components/form/
The Input Sizing paragraph

**Comment by denys-davydov** [comment]
@jonbwalker Actually, it is not an error, it is a warning. And the origin of this warning is unknown for now. 
```
warning.js:33 Warning: A component is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
    in input (created by Input)
    in Input (created by Domain)
    in div (created by Domain)
    in div (created by ExpansionPanelDetails)
```
Since the first line for the trace stake is ```in input (created by Input)``` I suspect it can be an issue with Input component which is a part od reactstrap library.

File: app/javascript/Application/components/Assessment/Assessment.js [hunk 4]
```diff
@@ -75,6 +75,19 @@ class Assessment extends Component {
     this.props.onAssessmentUpdate(assessment);
   };
 
+  updateCaregiverName = (caregiverIndex, caregiverName) => {
```
Comment (edgibbs) [inline app/javascript/Application/components/Assessment/Assessment.js]: for of replacement option, it worked fine for me when I swapped it:

```javascript
updateCaregiverName = (caregiverIndex, caregiverName) => {
     const assessment = cloneDeep(this.props.assessment);
     for (let domain of assessment.state.domains) {
       if (domain.caregiver_index === caregiverIndex) {
         domain.caregiver_name = caregiverName;
         break;
       }                                                                                                                                                                                                                                   
     }
     this.props.onAssessmentUpdate(assessment);
 };

```

**Pull request merged**